pkg/agent/nplagent/lib/utils.go

```diff
@@ -0,0 +1,87 @@
+// +build !windows
+
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lib
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"golang.org/x/sys/unix"
+	"k8s.io/klog"
+)
+
+// HasElem checks if a slice contains an element
+func HasElem(s interface{}, elem interface{}) bool {
+	arrV := reflect.ValueOf(s)
+
+	if arrV.Kind() == reflect.Slice {
+		for i := 0; i < arrV.Len(); i++ {
+			// panics if slice element points to an unexported struct field
+			// see https://golang.org/pkg/reflect/#Value.Interface
+			if arrV.Index(i).Interface() == elem {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+// ParsePortsRange parses port range and check if valid
+func ParsePortsRange(portRangeConfig string) (start, end int, err error) {
+	portsRange := strings.Split(portRangeConfig, "-")
+	if len(portsRange) != 2 {
+		return 0, 0, fmt.Errorf("wrong port range format: %s", portRangeConfig)
+	}
+
+	if start, err = strconv.Atoi(portsRange[0]); err != nil {
+		return 0, 0, err
+	}
+
+	if end, err = strconv.Atoi(portsRange[1]); err != nil {
+		return 0, 0, err
+	}
+
+	if end <= start {
+		return 0, 0, fmt.Errorf("invalid port range: %s", portRangeConfig)
+	}
+
+	return start, end, nil
+}
+
+// IsPortAvailable checks if a port is free or being used by any other process
+func IsPortAvailable(mPort int) bool {
```

Comment:
How about moving this one to pkg/agent/util/net_linux.go?

Reply:
Would you change net_linux.go to net_unix.go and compile it for non-Windows platforms?

Reply:
github.com/containernetworking/plugins/pkg/ns does not compile for macOS and it is used in net_linux.go. 
I get an error - build constraints exclude all Go files

Reply:
How about a separate net_unix.go just for this func?

Reply:
Yes. That would work. I'll update.